src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -1325,7 +1325,8 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 	private async doHandleExplorerDropOnMove(sources: ExplorerItem[], target: ExplorerItem): Promise<void> {
 
 		// Do not allow moving readonly items
-		const resourceFileEdits = sources.filter(source => !source.isReadonly).map(source => new ResourceFileEdit(source.resource, joinPath(target.resource, source.name)));
+		sources = sources.filter(s => !s.isReadonly);
+		const resourceFileEdits = await this.mergeDirectories(sources, target.resource);
```

Comment:
Should this be something the user is asked whether to merge or to replace?

Reply:
We can ask as that is one of the proposals in this issue. I feel like merging is pretty risk free, but it's not the default behavior on all OSes. On macOS it's only the default if you hold option whereas Windows it's always the default.